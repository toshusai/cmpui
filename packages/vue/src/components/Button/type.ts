export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
