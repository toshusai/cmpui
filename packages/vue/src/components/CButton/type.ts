export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
