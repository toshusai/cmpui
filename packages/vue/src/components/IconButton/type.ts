export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
