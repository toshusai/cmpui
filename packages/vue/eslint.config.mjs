"use strict";
import pluginVue from "eslint-plugin-vue";
import eslintConfigPrettier from "eslint-config-prettier";
import tseslint from "typescript-eslint";

export default [
  ...tseslint.configs.recommended,
  ...pluginVue.configs["flat/recommended"],
  {
    ignores: ["dist/*", "types/*"],
  },
  {
    rules: {
      "vue/multi-word-component-names": "off",
      "no-unused-vars": "off",
      "@typescript-eslint/no-unused-vars": ["error"],
    },
    languageOptions: {
      parserOptions: { parser: tseslint.parser, sourceType: "module" },
    },
  },
  eslintConfigPrettier,
];
