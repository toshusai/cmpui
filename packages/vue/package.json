{
  "name": "@toshusai/cmpui-vue",
  "version": "1.0.0-alpha.8",
  "description": "CmpUI is a component library designed for creating complex GUI applications in web browsers.",
  "main": "./dist/index.umd.js",
  "module": "./dist/index.es.js",
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "import": {
        "default": "./dist/index.es.js",
        "types": "./dist/index.d.ts"
      },
      "require": {
        "default": "./dist/index.umd.js",
        "types": "./dist/index.d.ts"
      }
    }
  },
  "scripts": {
    "typecheck": "vue-tsc -p tsconfig.ci.json",
    "test:watch": "vitest",
    "test": "vitest run",
    "build": "npm run typecheck && vite build && vue-tsc",
    "storybook": "storybook dev -p 6006",
    "build-storybook": "storybook build",
    "format": "prettier --write .",
    "lint": "eslint --fix .",
    "fix": "npm run format && npm run lint"
  },
  "dependencies": {
    "@toshusai/cmpui-css": "link:../css",
    "vue": "^3.4.38"
  },
  "devDependencies": {
    "@chromatic-com/storybook": "^1.7.0",
    "@storybook/addon-essentials": "8.2.6",
    "@storybook/addon-interactions": "8.2.6",
    "@storybook/addon-links": "8.2.6",
    "@storybook/addon-mdx-gfm": "8.2.6",
    "@storybook/addon-onboarding": "8.2.6",
    "@storybook/addon-themes": "8.2.6",
    "@storybook/blocks": "8.2.6",
    "@storybook/test": "8.2.6",
    "@storybook/test-runner": "^0.17.0",
    "@storybook/theming": "8.2.6",
    "@storybook/vue3": "^8.2.9",
    "@storybook/vue3-vite": "^8.2.9",
    "@types/jest": "^29.5.11",
    "@vitejs/plugin-vue": "^5.1.2",
    "@vue/compiler-sfc": "^3.4.38",
    "concurrently": "^8.2.2",
    "eslint": "^9.9.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-vue": "^9.27.0",
    "storybook": "8.2.6",
    "typescript": "^5.2.2",
    "typescript-eslint": "^8.2.0",
    "vitest": "^1.1.0",
    "vue-tsc": "^2.0.29"
  },
  "keywords": [
    "typescript",
    "components",
    "ui",
    "gui"
  ],
  "author": "toshusai",
  "license": "MIT"
}
