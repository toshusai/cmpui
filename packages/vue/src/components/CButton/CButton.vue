<script setup lang="ts">
import { CButtonProps } from "./type";

withDefaults(defineProps<CButtonProps>(), {
  variant: "primary",
  size: "M",
  disabled: false,
});
</script>

<template>
  <button
    class="cmpui_button"
    :data-variant="variant"
    :data-size="size"
    :disabled="disabled"
    v-bind="$attrs"
  >
    <slot></slot>
  </button>
</template>
