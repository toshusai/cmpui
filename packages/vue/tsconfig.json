{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "module": "ESNext",
    "skipLibCheck": true,

    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "emitDeclarationOnly": true,
    "declaration": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "react-jsx",
    "outDir": "dist",

    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,
    "typeRoots": ["node_modules/@types", ".storybook/@types", "./node_modules"],
    "types": ["@vitest/browser/providers/playwright"]
  },
  "include": ["src"],
  "exclude": ["src/**/*.stories.ts"]
}
