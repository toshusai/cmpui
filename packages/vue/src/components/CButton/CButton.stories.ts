import type { Meta, StoryObj } from "@storybook/vue3";
import CButton from "./CButton.vue";
import { h } from "vue";

const meta = {
  title: "inputs/CButton",
  tags: ["autodocs"],
  component: CButton,
} satisfies Meta<typeof CButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    default: "Button",
    variant: "primary",
  },
};

export const Variants: Story = {
  render: () =>
    h(() => [
      h(CButton, { variant: "primary" }, () => "Primary"),
      h(CButton, { variant: "secondary" }, () => "Secondary"),
      h(CButton, { variant: "transparent" }, () => "Transparent"),
    ]),
};

export const Sizes: Story = {
  render: () =>
    h(() => [
      h(CButton, { size: "M" }, () => "Medium"),
      h(CButton, { size: "S" }, () => "Small"),
    ]),
};

export const Disabled: Story = {
  render: () =>
    h(() => [
      h(CButton, { variant: "primary", disabled: true }, () => "Primary"),
      h(CButton, { variant: "secondary", disabled: true }, () => "Secondary"),
    ]),
};

export const Bordered: Story = {
  render: () =>
    h(() => [
      h(CButton, { variant: "primary", bordered: true }, () => "Primary"),
      h(CButton, { variant: "secondary", bordered: true }, () => "Secondary"),
    ]),
};
