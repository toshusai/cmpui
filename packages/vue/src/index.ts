import CButton from "./components/CButton/CButton.vue";

export { CButton };
