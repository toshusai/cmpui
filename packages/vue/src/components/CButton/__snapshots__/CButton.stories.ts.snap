// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CButton Default smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="M"
        default="Button"
>
  Button
</button>
`;

exports[`components/CButton Disabled smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="M"
        disabled
        default="Button"
>
  Button
</button>
`;

exports[`components/CButton Secondary smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="secondary"
        data-size="M"
        default="Button"
>
  Button
</button>
`;

exports[`components/CButton Small smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="S"
        default="Button"
>
  Button
</button>
`;
